docs/src/webview2.md

```diff
@@ -0,0 +1,128 @@
+BrowserContext context = browser.contexts().get(0);
+Page page = context.pages().get(0);
+```
+
+```python async
+browser = await playwright.chromium.connect_over_cdp("http://localhost:9222")
+context = browser.contexts[0]
+page = context.pages[0]
+```
+
+```python sync
+browser = playwright.chromium.connect_over_cdp("http://localhost:9222")
+context = browser.contexts[0]
+page = context.pages[0]
+```
+
+```csharp
+var browser = await playwright.Chromium.ConnectOverCDPAsync("http://localhost:9222");
+var context = browser.Contexts[0];
+var page = context.Pages[0];
+```
+
+To ensure that the WebView2 control is ready, you can wait for the [`CoreWebView2InitializationCompleted`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.wpf.webview2.corewebview2initializationcompleted?view=webview2-dotnet-1.0.1343.22) event:
+
+```txt
+this.webView.CoreWebView2InitializationCompleted += (_, e) =>
+{
+    if (e.IsSuccess)
+    {
+        Console.WriteLine("WebView2 initialized");
+    }
+};
+```
+
+## Writing and running tests
+
+By default, the WebView2 control will use the same user data directory for all instances. This means that if you run multiple tests in parallel, they will infer with each other. To avoid this, you can set the `WEBVIEW2_USER_DATA_FOLDER` environment variable (or [WebView2.EnsureCoreWebView2Async Method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.wpf.webview2.ensurecorewebview2async?view=webview2-dotnet-1.0.1343.22) method) to a different folder for each test. This will make sure that each test runs in its own user data directory:
+
+Using the following base test instance, Playwright will run your WebView2 application as a sub-process and assigns a unique user data directory to it. For that create a `webView2Test.ts` inside your `tests` directory:
+
+```ts
+import { test as base } from '@playwright/test';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import childProcess from 'child_process';
+
+export const test = base.extend({
+  browser: async ({ playwright }, use, testInfo) => {
+    const cdpPort = 10000 + testInfo.workerIndex;
+    const executable = path.join(__dirname, '../webview2-app/bin/Debug/net6.0-windows/webview2.exe');
+    fs.accessSync(executable, fs.constants.X_OK); // Make sure that the executable exists and is executable
+    const userDataDir = path.join(fs.realpathSync.native(os.tmpdir()), `playwright-webview2-tests/user-data-dir-${testInfo.workerIndex}`);
+    const webView2Process = childProcess.spawn(executable, [], {
+      shell: true,
+      env: {
+        ...process.env,
+        WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS: `--remote-debugging-port=${cdpPort}`,
+        WEBVIEW2_USER_DATA_FOLDER: userDataDir,
+      }
+    });
+    await new Promise<void>(resolve => webView2Process.stdout.on('data', data => {
+      if (data.toString().includes('WebView2 initialized'))
```

Comment:
This needs a comment explaining that it's our app that prints the message to stdout.